modules/core/src/loaders/xviz-file-loader.js

```diff
@@ -73,26 +75,29 @@ export default class XVIZFileLoader extends XVIZLoaderInterface {
     return this._loadFile(metadataPath, {worker: false});
   }
 
-  _loadNextBatch(startFrame) {
+  _startLoad() {
+    // fetching in parallel
+    for (let i = 0; i < this._batchSize && i < this._numberOfFrames; i++) {
+      this._loadNextFrame();
+    }
+  }
+
+  _loadNextFrame() {
     if (!this.isOpen()) {
       return;
     }
 
-    if (startFrame >= this._numberOfFrames) {
+    this._lastLoadFrame = this._lastLoadFrame + 1;
```

Comment:
Nit: `this._lastLoadFrame` should only be bumped if the frame actually gets loaded.

Reply:
I intended to bump `_lastLoadFrame` before promised is resolved, because the same frame might be fetched more than once if bumped after.